src/lib/server/client_connection.hpp

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include <boost/asio/ip/tcp.hpp>
+#include <boost/thread/future.hpp>
+
+#include <memory>
+
+namespace opossum {
+
+using boost::asio::ip::tcp;
```

Comment:
I don't like usings for third party libs, makes it harder to determine where `tcp::socket socket` comes from